### ISSUE TITLE
[Dotcom shell]: React VO does not announce location to user after using Skip to content button

### ISSUE THREAD

**Issue by oliviaflory** [carbon-design-system/carbon-for-ibm-dotcom#8093]
### Description

When the user selects the Skip to content button from the masthead, the screen reader does not announce the new location. User must tab to get the VO to announce anything, skipping directly to the first TOC item,  and misses the "You are in a list" announcement.

https://user-images.githubusercontent.com/45692622/150417561-321e43d6-ee8c-478d-aa37-203172e095e1.mov

 _Present in Production_

### Component(s) impacted

Masthead (assuming it affects masthead)
Dotcom shell

### Browser

Chrome

### Carbon for IBM.com version

1.30.0 rc 0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

carbon for IBM.com

### Package

@carbon/ibmdotcom-react

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/react/?path=/story/components-dotcom-shell--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by jeffchew** [comment]
Need to ask accessibility team for guidance on this.

**Comment by ljcarot** [comment]
Try testing on codesandbox...if still exists there, assign severity and pipeline. @oliviaflory

**Comment by oliviaflory** [comment]
Retested in both React production and React 1.31.0 rc1 

**React v1.30**
When I select the Skip to main content button the VO does not announce where I am within the main content. 
https://user-images.githubusercontent.com/45692622/154540146-a9229a23-7a84-4dd4-80c2-c23bce7eaae4.mov

**React v1.31.0 rc 1**
https://user-images.githubusercontent.com/45692622/154540101-c61df891-cfc8-45e0-98a5-3ef64846cb5d.mov



Web components v1.16.0 rc1
Web components does not have Skip to main content button enabled
I believe all navigation needs a Skip to main content option if the user does not wish to go through the main navigation.

https://user-images.githubusercontent.com/45692622/154540215-1bd3960a-b931-4b6d-94cc-5887fd837e96.mov

**Comment by oliviaflory** [comment]
> @oliviaflory I'm not sure it needs to announce the main content area, based on the accessibility documentation. https://www.carbondesignsystem.com/components/UI-shell-header/usage/#interactions https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G1
> 
> Since the link itself already announces it is for skipping to the main content, it seems this is behaving correctly?

@kennylam hmm ok, I guess I assumed you'd want to hear what you landed on, but to does sound like we might be ok to close.

I swear there was another example that announced "you are currently on a list" when you land on the TOC, but I don't hear it in that example
